pkg/asset/machines/vsphere/capimachines.go

```diff
@@ -57,7 +61,7 @@ func getNetworkInventoryPath(vcenterContext vsphere.VCenterContext, networkName
 
 // GenerateMachines returns a list of capi machines.
 func GenerateMachines(ctx context.Context, clusterID string, config *types.InstallConfig, pool *types.MachinePool, osImage string, role string, metadata *vsphere.Metadata) ([]*asset.RuntimeFile, error) {
-	machines, _, err := Machines(clusterID, config, pool, osImage, role, "")
+	machines, _, _, _, err := Machines(clusterID, config, pool, osImage, role, "") //nolint:dogsled
```

Comment:
can the code be refactored such that this nolint isnt needed or is there a firm reason to keep it?

Reply:
If it gets refactored, do as separate PR.  The CAPI env was not what the Machine interface was intended for.  I am not sure why it was reused instead of consuming what may have been generated in the normal machine creation flow.

Reply:
This has now been refactored and nolint has been removed.